From d6cd80e4efda43e9206aec46336a3d2ab21a0a36 Mon Sep 17 00:00:00 2001
From: Gerald Yang <gerald.yang@canonical.com>
Date: Thu, 7 Jan 2021 03:04:02 +0000
Subject: [PATCH] Remove logging every slow request details to monitors

 recent change https://tracker.ceph.com/issues/43975 was made to slow
 request logging to include detail on each operation in the cluster logs.
 With this change, detail for every slow request is always sent to the
 monitors and added to the cluster logs.

 This does not scale. Large, high-throughput clusters can overwhelm
 their monitors with spurious logs in the event of a performance issue.
 Disrupting the monitors can then cause further instability in the
 cluster.

 This SRU reverts the cluster logging of every slow request the osd is
 processing.

 The slow request clog change was added in nautilus (14.2.10) and
 octopus (15.2.0).

Signed-off-by: Gerald Yang <gerald.yang@canonical.com>
---
 src/osd/OSD.cc | 1 -
 1 file changed, 1 deletion(-)

Bug-Ubuntu: https://bugs.launchpad.net/bugs/1909162

Index: ceph/src/osd/OSD.cc
===================================================================
--- ceph.orig/src/osd/OSD.cc
+++ ceph/src/osd/OSD.cc
@@ -7780,7 +7780,6 @@ vector<DaemonHealthMetric> OSD::get_heal
            << " currently "
            << op.state_string();
         lgeneric_subdout(cct,osd,20) << ss.str() << dendl;
-        clog->warn() << ss.str();
 	slow++;
 	if (!oldest_op || op.get_initiated() < oldest_op->get_initiated()) {
 	  oldest_op = &op;
